src/main/kotlin/solve/scene/SceneFacade.kt

```diff
@@ -34,6 +38,12 @@ class SceneFacade(private val controller: SceneController) {
             visualizationLayers[projectLayer.key] ?: projectLayer.toLayerSettings()
                 .also { visualizationLayers[projectLayer.key] = it }
         }
+
+        if (!keepSettings) {
```

Comment:
It is not required, keep settings flag is about scene settings, not layers settings